gdal/ogr/ogr_feature.h

```diff
@@ -79,6 +79,7 @@ class OGRStyleTable;
  * Definition of an attribute of an OGRFeatureDefn. A field is described by :
  * <ul>
  * <li>a name. See SetName() / GetNameRef()</li>
+ * <li>an alias. See SetAlias() / GetAliasRef()</li>
```

Comment:
I was wondering if we shouldn't generalize the name a bit. From https://desktop.arcgis.com/en/arcmap/latest/manage-data/tables/field-properties-aliases-and-table-display-options.htm , "An alias is an alternative name for a field—a more user-friendly description of the content of the field".  So I was thinking to "Description" which could potentially cover more drivers. The issue I see with "alias" is that one could expect it to be usable as a replacement for the name in SQL expressions for example.
But I don't have a strong opinion about that. Just wanted to raise the discussion

KML has a "displayName" that could potentially fit into that : https://developers.google.com/kml/documentation/extendeddata#declare-the-schema-element (although KML displayName can includes HTML tags)

Reply:
Hmm - my concern with equating this to description is that a description is likely to be lengthy, while an alias is somewhat in between the field name and a description. E.g. if we use description, then you could argue that field comments from postgres should be supported. But I've only ever seen them used for lengthy descriptions which would be unsuitable for an "alias". What about "AlternativeName"?

Reply:
Or "Extended/expanded" name?

Reply:
ExtendedName would be fine to me. In the comments you could add what you mentioned above, that is that is expected to be a longer form of Name but not a too lengthy description / comment. And that it for example maps to FileGeodabase alias.
Is the purpose of using them in the header line of QGIS attribute table or edit forms ?

Reply:
> I would have voted for "AlternativeName".

I'd be fine with AlternativeName, but just seeing that https://en.wikipedia.org/wiki/Alias defines  "Alias, an alternative name" :-) So perhaps alias is just fine.

Reply:
> Is the purpose of using them in the header line of QGIS attribute table or edit forms ?

Yes -- it's so that we can automatically set the field alias in qgis to match those defined in the data source. The ultimate outcome is that official datasets published with cryptic field names should be a bit more usable!

> I'd be fine with AlternativeName

Let's go with that -- it avoids the confusing wrt SQL aliases.